src/providers/journal/__tests__/journal.spec.ts

```diff
@@ -6,44 +6,117 @@ import { AuthenticationProviderMock } from '../../authentication/__mocks__/authe
-          { provide: NetworkStateProvider, useClass: NetworkStateProviderMock },
-          { provide: SecureStorage, useClass: SecureStorageMock },
-          { provide: Network, useClass: NetworkMock },
-        ],
+  let journalProvider: JournalProvider;
+  let httpMock: HttpTestingController;
+  let authProviderMock: AuthenticationProvider;
+  let urlProviderMock: UrlProvider;
+  let dataStoreMock: DataStoreProvider;
+  let appConfigProviderMock: AppConfigProvider;
+  let cacheDays: number;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        HttpClientTestingModule,
+      ],
+      providers: [
+        JournalProvider,
+        { provide: UrlProvider, useClass: UrlProviderMock },
+        { provide: AuthenticationProvider, useClass: AuthenticationProviderMock },
+        { provide: DataStoreProvider, useClass: DataStoreProviderMock },
+        { provide: NetworkStateProvider, useClass: NetworkStateProviderMock },
+        { provide: SecureStorage, useClass: SecureStorageMock },
+        { provide: Network, useClass: NetworkMock },
+        { provide: AppConfigProvider, useClass: AppConfigProviderMock },
+      ],
+    });
+
+    httpMock = TestBed.get(HttpTestingController);
+    journalProvider = TestBed.get(JournalProvider);
+    authProviderMock = TestBed.get(AuthenticationProvider);
+    urlProviderMock = TestBed.get(UrlProvider);
+    dataStoreMock = TestBed.get(DataStoreProvider);
+    appConfigProviderMock = TestBed.get(AppConfigProvider);
+    cacheDays = appConfigProviderMock.getAppConfig().daysToCacheJournalData;
+
+  });
+
+  describe('isCacheTooOld', () => {
+    it(`should return true if date is greater than cacheDays days ago`, () => {
+      const today = new DateTime();
+      const tooOld = new DateTime().add(-(cacheDays + 1), Duration.DAY);
+      expect(journalProvider.isCacheTooOld(tooOld, today)).toBe(true);
+    });
+
+    it(`should return true if date is less than or equal to cacheDays days ago`, () => {
+      const today = new DateTime();
+      const withinWindow = new DateTime().add(-cacheDays, Duration.DAY);
+      expect(journalProvider.isCacheTooOld(withinWindow, today)).toBe(false);
+    });
+  });
+
+  describe('getAndConvertOfflineJournal', () => {
+    it('should empty cached data if cache is too old', () => {
+      const exampleSchedule: ExaminerWorkSchedule = {
+        examiner: { staffNumber: '1234' },
+      };
+      const agedCache = {
+        dateStored: new DateTime().add(-(cacheDays + 1), Duration.DAY).format('YYYY/MM/DD'),
+        data: exampleSchedule,
+      };
+
+      // override mock getItem as we need data to test
+      // @ts-ignore
+      dataStoreMock.getItem.and.callFake(() => Promise.resolve(JSON.stringify(agedCache)));
+
+      spyOn(journalProvider, 'emptyCachedData').and.callThrough();
+
+      journalProvider.getAndConvertOfflineJournal().then((data) => {
+        expect(journalProvider.emptyCachedData).toHaveBeenCalled();
```

Comment:
Do these tests get called? We don't have `async` or `done()` being called here..

Reply:
Ignore me, I think these are ok actually as they're wrapped in a `.then`